### DISCUSSION TITLE
Question: difference betweem cat rows and rowskey

### DISCUSSION THREAD

**Discussion by fgvieira** [dathere/qsv#1594] [FAQ]
Not quite sure I undestand the difference between `cat rows` and `cat rowskey`.
Isn't `qsv cat rows` == `qsv cat rowskey --group none`?

**Comment by a5dur** [comment]
Great question! Let's clarify that:

They do achieve similar results, but there's a subtle difference in how they concatenate and treat grouping.

`cat rows` assumes that all CSV files have the same structure (same columns in the same order).
While `rowskey` concatenates rows based on a specified key or column. It identifies a common key column in all csv files and merges the rows that share the same key. the group option introduces a grouping column indicating the source of each row.

So, technically, you are correct: `qsv cat rows` is equivalent to `qsv cat rowskey --group none`. 
Both concatenate rows without introducing a special grouping column... 

but the `--group` option in `qsv cat rowskey` becomes more relevant when you want to customize the grouping behavior, for example, grouping by the full file path, filename, or other options. If --group none is used, it essentially behaves like qsv cat rows.

I hope this helps clarify the difference! also qsv also has a very [user-friendly website](qsv.dathere.com/web/cat) where you could learn more and test out this command

**Comment by jqnatividad** [comment]
Thanks @a5dur for the expansive answer!  Moving this to the Discussion area's FAQ section so other folks with the same query can find it easily.
